allennlp/models/biaffine_dependency_parser_multilang.py

```diff
@@ -0,0 +1,296 @@
+from typing import Dict, Optional, Any, List
+import logging
+
+from collections import defaultdict
+from overrides import overrides
+import torch
+import numpy
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.data import Vocabulary
+from allennlp.modules import Seq2SeqEncoder, TextFieldEmbedder, Embedding
+from allennlp.modules import FeedForward
+from allennlp.models.model import Model
+from allennlp.models.biaffine_dependency_parser import BiaffineDependencyParser
+from allennlp.nn import InitializerApplicator, RegularizerApplicator
+from allennlp.nn.util import get_text_field_mask
+from allennlp.training.metrics import AttachmentScores
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@Model.register("biaffine_parser_multilang")
+class BiaffineDependencyParserMultiLang(BiaffineDependencyParser):
```

Comment:
You're inheriting from the above parser and not really adding anything to the constructor, but you're duplicating a whole lot of logic from the `forward` method (and reformatting it, so it's not as obvious that it's copied).  Can you just factor out the pieces that you need in the superclass to separate methods, and call those, instead of duplicating all of this logic?

Reply:
done 👍